File: clients/fsharp/Model/Player.fs
#nowarn "0058"

namespace SpbAiChamp
namespace SpbAiChamp.Model

open SpbAiChamp

/// Player (game participant)
type Player = {
    /// Current score points
    Score: int;
} with

    /// Write Player to writer
    member this.writeTo(writer: System.IO.BinaryWriter) =
        writer.Write this.Score
        ()

    /// Read Player from reader
    static member readFrom(reader: System.IO.BinaryReader) = {
        Score = reader.ReadInt32()
    }

File: clients/rust/src/model/player.rs
use super::*;

/// Player (game participant)
#[derive(Clone, Debug, PartialEq, Eq, Hash)]
pub struct Player {
    /// Current score points
    pub score: i32,
}

impl trans::Trans for Player {
    fn write_to(&self, writer: &mut dyn std::io::Write) -> std::io::Result<()> {
        self.score.write_to(writer)?;
        Ok(())
    }
    fn read_from(reader: &mut dyn std::io::Read) -> std::io::Result<Self> {
        let score: i32 = trans::Trans::read_from(reader)?;
        Ok(Self {
            score,
        })
    }
}